### ISSUE TITLE
VideoReady Event not firing on Windows Build?

### ISSUE THREAD

**Issue by Kuinshi** [RenderHeads/UnityPlugin-AVProVideo#321]
Problem description:

Hey,
for a current project I need to run 4 transparent FullHD Videos at the same time, which obviously has bad performance. Thus I tried waiting for the Video Ready Event, which works kind of fine in the Unity Editor. But when I build my Project to Windows, the Event is not firing at all and thus the video is not playing at all...
Is this a known issue? And if yes is there any way to fix it or work around it? (If set the video to autoplay it does play on a Windowsbuild, so problem is with the event.)

Also I'd be happy for any other tips, that could help increase the performance of running 4 transparent FullHD Videos at the same time, the videos do seem to stutter and flicker lightly even when waiting for the ready event.

Thanks! :)
If there 

Device (which devices are you having the issue with - model, OS version number):

Windows 10 PC

Media (tell us about your videos - number of videos, resolution, codec, frame-rate, example URLs):

Four Videos; Transparent FullHD, so FullHD twice; H264; 25fps

System Information:

AVPro Video: plugin v1.10.2 scripts v1.10.3
Target Platform: Standalone
Unity: v2019.3.2f1 WindowsEditor
OS: Desktop - To Be Filled By O.E.M. (To Be Filled By O.E.M.) - Windows 10  (10.0.0) 64bit - English
CPU: AMD Ryzen 5 1600 Six-Core Processor  - 12 threads - 16316KB
GPU: NVIDIA GeForce GTX 1060 3GB - NVIDIA - Direct3D 11.0 [level 11.1] - 2988KB - 16384

**Comment by AndrewRH** [comment]
Hi,

Thanks for getting in touch.

I will answer your issue in the next comments, but here are some thoughts about smoother playback.

For smoother playback, I would recommend authoring the videos at 30fps instead of 25.  25 is not a multiple of your display device (probably), so it will cause frame stutters.  

You should also check the Task Manager Performance tab to see if there is an element of your system that's causing a bottleneck.  The GPU video decoder may be overloaded for example.

It's also possible to author videos in a way that helps playback - eg using simpler encoding options, less B/Ref frames and even some "fastdecode" tuning profiles.   Finally, depending on your use case, you could use a codec like Hap Alpha instead to achieve transparency - though it has massive file sizes.

We also have some sections in the documentation about how to encode videos for better playback.

Thanks,

**Comment by AndrewRH** [comment]
Hi,

Thanks for getting in touch.

I will answer your issue in the next comments, but here are some thoughts about smoother playback.

For smoother playback, I would recommend authoring the videos at 30fps instead of 25.  25 is not a multiple of your display device (probably), so it will cause frame stutters.  

You should also check the Task Manager Performance tab to see if there is an element of your system that's causing a bottleneck.  The GPU video decoder may be overloaded for example.

It's also possible to author videos in a way that helps playback - eg using simpler encoding options, less B/Ref frames and even some "fastdecode" tuning profiles.   Finally, depending on your use case, you could use a codec like Hap Alpha instead to achieve transparency - though it has massive file sizes.

We also have some sections in the documentation about how to encode videos for better playback.

Thanks,

**Comment by AndrewRH** [comment]
Do you mean the "ReadyToPlay" event?  I just ran some tests here and it's firing fine for me...Are you doing IL2CPP builds or anything else fancy?  Are you building to 32-bit or 64-bit Windows Standalone?

The ReadyToPlay event should only fire when you have AutoPlay option disabled.

**Comment by Kuinshi** [comment]
Hi,

thanks for the quick answer, did not expect it.

We were using transparent 4k Videos first, those resulted in 100% CPU Usage, even though I checked that the hardware decoding is ticked - I guess my PC's CPU is a bit of a bottleneck. Though swapping to the fullHD Video only results in around 60% CPU and GPU usage, so that shouldnt be the issue anymore. We will try the FPS and Encoding-suggestions, thanks! :)



> Do you mean the "ReadyToPlay" event? I just ran some tests here and it's firing fine for me...Are you doing IL2CPP builds or anything else fancy? Are you building to 32-bit or 64-bit Windows Standalone?
> 
> The ReadyToPlay event should only fire when you have AutoPlay option disabled.

Yes I did indeed mean the ReadyToPlay event. I am building on 64-bit Windows standalone, but I am using Mono instead of IL2CPP. Trying to switch to IL2CPP results in a compiler error which I dont see a quick fix for. I will try in a separate smaller scene if that could be the issue though.
